core/application/bloodDonationWorkflow/BloodDonationService.ts

```diff
@@ -0,0 +1,58 @@
+import { GENERIC_CODES } from '../../../commons/libs/constants/GenericCodes'
+import BloodDonationOperationError from './BloodDonationOperationError'
+import { DonationDTO } from '../../../commons/dto/DonationDTO'
+import { generateUniqueID } from '../utils/idGenerator'
+import Repository from '../technicalImpl/policies/repositories/Repository'
+import { generateGeohash } from '../utils/geohash'
+import { validateInputWithRules } from '../utils/validator'
+import { BloodDonationAttributes, validationRules, UpdateBloodDonationAttributes } from './Types'
+
+export class BloodDonationService {
+  async createBloodDonation(donationAttributes: BloodDonationAttributes, bloodDonationRepository: Repository<DonationDTO>): Promise<string> {
+    try {
+      const validationResponse = validateInputWithRules({ bloodQuantity: donationAttributes.bloodQuantity, donationDateTime: donationAttributes.donationDateTime }, validationRules)
+      if (validationResponse !== null) {
+        return validationResponse
+      }
+      await bloodDonationRepository.create({
+        id: generateUniqueID(),
+        ...donationAttributes,
+        status: 'accepted',
```

Comment:
why are we using hardcoded values when you've declared an enum

Reply:
Create an enum and use it in code. [Commit](https://github.com/CraftsmenLtd/BloodConnect/pull/177/commits/be92e1051e74df3cdc6edffb1aad6cfb725c40e8)